### PR TITLE
Add set text analyser for HTML lexer

### LINKED ISSUE #94: [Chroma] Add text analyzer for HTML and XML

### PR DESCRIPTION
**Pull request by gandarez**
This PR ports pygments HTML text analysis to chroma/go. Original code can be found at: https://github.com/pygments/pygments/blob/master/pygments/lexers/html.py#L94

It also creates a new package called `doctype` which contains a regex for matching html stuff.

Fixes: wakatime/wakatime-cli#94

### CODE REVIEW

File: lexers/h/html.go [hunk 1]
```diff
@@ -56,4 +57,11 @@ var HTML = internal.Register(MustNewLexer(
 			{`[^\s>]+`, LiteralString, Pop(1)},
 		},
 	},
-))
+).SetAnalyser(func(text string) float32 {
+	matched, _ := doctype.MatchString(text, "html")
+	if matched {
```
Comment (dron22) [inline lexers/h/html.go]: You can do this in one line
```suggestion
	if matched, _ := doctype.MatchString(text, "html"); matched {
```

File: pkg/doctype/doctype.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var doctypeLookupRe = regexp.MustCompile(`(?ms)(<\?.*?\?>)?\s*<!DOCTYPE\s+([a-zA-Z_][a-zA-Z0-9]*(?: \s+[a-zA-Z_][a-zA-Z0-9]*\s+"[^"]*")?)[^>]*>`)
+
+// MatchString check if the doctype matches a regular expression (if present).
+func MatchString(text string, pattern string) (bool, error) {
```
Comment (dron22) [inline pkg/doctype/doctype.go:12]: As it is an exported function, can we have a test for this one as well?

**Review by dron22 — CHANGES_REQUESTED** [review]

File: pkg/doctype/doctype.go (hunk 2 above)
Comment (gandarez) [reply pkg/doctype/doctype.go:12]: Forgot to add them.

**Comment by gandarez** [comment]
Fixes applied.

File: pkg/doctype/doctype_test.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+			Pattern:  `html.*`,
+			Expected: true,
+		},
+		"full html match": {
+			Text:     "<?xml ?><!DOCTYPE html PUBLIC  \"-//W3C//DTD XHTML 1.0 Strict//EN\">",
+			Pattern:  `html`,
+			Expected: true,
+		},
+		"xml not match": {
+			Text:     "<?xml ?> <DOCTYPE html PUBLIC \"a\"> <html>",
```
Comment (dron22) [inline pkg/doctype/doctype_test.go:28]: This is not matching because of missing `!`. Not because it is an xml Docstring. Is this what you intended?

Comment (gandarez) [reply pkg/doctype/doctype_test.go:28]: Tests were copied from this piece of code https://github.com/pygments/pygments/blob/master/tests/test_util.py#L115

Comment (dron22) [reply pkg/doctype/doctype_test.go:28]: Ok cool. Then let's rename test description from `xml not match` to `missing exclamation mark`

**Review by dron22 — APPROVED** [review]
Approving already, just test description adjustment left.

**Pull request merged**